mettle/src/stdapi/ui/ui.c

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2015 Rapid7
+ * @brief UI API
+ * @file ui.c
+ */
+
+#include <endian.h>
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <mettle.h>
+
+#include "channel.h"
+#include "log.h"
+#include "tlv.h"
+
+#if __APPLE__
+#include <TargetConditionals.h>
+#endif
+
+#if TARGET_OS_OSX
+extern struct tlv_packet *desktop_screenshot(struct tlv_handler_ctx *ctx);
+#endif
+
+void ui_register_handlers(struct mettle *m)
+{
+	struct tlv_dispatcher *td = mettle_get_tlv_dispatcher(m);
+
+#if TARGET_OS_OSX
```

Comment:
Same as the Linux camera support, I'd surround this by a feature flag that's set at configure time than an OS flag. Other than that, looks good to me!